docs/data/joy/components/modal/modal.md

```diff
@@ -8,113 +8,139 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/
+The `ModalClose` component accepts the `variant` property, since it shares the same styles as [`IconButton`](/joy-ui/react-button/#icon-button).
 :::
 
-### Close reason
+### Sizes
 
-The second argument of the `onClose` gives you the information about how the event is triggered.
+The `ModalDialog` component offers three distinct sizing options: `sm`, `md` (default), and `lg`. The `ModalClose` and `ModalDialogTitle` components will automatically inherit the size set in `ModalDialog`, unless explicitly specified within each individual component.
 
-The possible values are:
+{{"demo": "SizeModalDialog.js"}}
 
-- `backdropClick`: the user clicks on the modal's backdrop.
-- `escapeKeyDown`: the user presses `Escape` on the keyboard.
-- `closeClick`: the user clicks on the `ModalClose` element.
+### Colors
 
-{{"demo": "CloseModal.js"}}
+Every palette included in the theme is available via the `color` prop.
 
-### Modal Dialog
+### Transition
 
-To create a modal dialog, render the Modal Dialog component inside the Modal.
+The modal components **do not** come with built-in transitions.
+To incorporate animations, you will need to create your own transitions or use an external library.
 
-The Dialog will apply spacing to the elements that have `aria-labelledby` or `aria-describedby` attribute.
+The following is an example using [`react-transition-group`](https://reactcommunity.org/react-transition-group/transition/) to create a fade animation:
 
-{{"demo": "BasicModalDialog.js"}}
+{{"demo": "FadeModalDialog.js"}}
 
-#### Variant
+### Nesting modals
 
-The Modal Dialog supports the [global variants](/joy-ui/main-features/global-variants/) feature.
+Modal components in Joy UI are able to be nested:
 
-The Modal Close component's variant adapts automatically to contrast with the Modal Dialog, as demonstrated below:
+{{"demo": "NestedModals.js"}}
 
-{{"demo": "VariantModalDialog.js"}}
+:::warning
+While it is possible to create nested modals, stacking more than two at a time is discouraged.
+Each additional modal further restricts interactions with elements in the background, which renders the previous states inaccessible and complicating the user experience.
+:::
 
-#### Size
+## Usage with Modal Close
 
-The Modal Dialog comes in 3 sizes: `sm`, `md` (default), and `lg`.
+```tsx
+import ModalClose from '@mui/joy/ModalClose';
+```
 
-The Modal Close and Modal Dialog Title components inherit the size from the Modal Dialog unless specified in each component directly.
+Use the `ModalClose` component to render a close button that automatically adopts the modal's `onClose` function.
+The second argument of the `onClose` gives information about how the event is triggered.
 
-{{"demo": "SizeModalDialog.js"}}
+The potential triggers are:
+
+- `backdropClick`: the user clicks on the modal's backdrop.
+- `escapeKeyDown`: the user presses `Escape` on the keyboard.
```

Comment:
You can use this nifty lil `<kbd class="key">` tag to give it keyboard key styles :)

```suggestion
- `escapeKeyDown`: the user presses <kbd class="key">Escape</kbd>.
```